metecho/api/jobs.py

```diff
@@ -581,7 +582,30 @@ def delete_scratch_org(scratch_org, *, originating_user_id):
 
 
 def refresh_github_repositories_for_user(user):
-    user.refresh_repositories()
+    from .models import GitHubRepository
+
+    try:
+        repos = get_all_org_repos(user)
+        with transaction.atomic():
+            GitHubRepository.objects.filter(user=user).delete()
+            GitHubRepository.objects.bulk_create(
+                [
+                    GitHubRepository(
+                        user=user,
+                        repo_id=repo.id,
+                        repo_url=repo.html_url,
+                        permissions=repo.permissions,
+                    )
+                    for repo in repos
+                ]
+            )
+    except Exception as e:
+        user.finalize_refresh_repositories(error=e)
+        tb = traceback.format_exc()
+        logger.error(tb)
+        raise
```

Comment:
Why are you explicitly logging the traceback before re-raising? The traceback for the raised exception should end up getting recorded in the failed rq job.

Reply:
I'm following the pattern from the other jobs, they all seem to do that. Perhaps the intention is to get a formatted traceback in the logs?

Reply:
I guess I have no problem with it if it's helpful. For myself, I'm always investigating after the fact when I have access to failed rq jobs but not old Heroku logs